docs/elements.md

```diff
@@ -1994,6 +1995,51 @@ The `pl-card` attributes mirror the options of [Bootstrap 4 cards](https://getbo
+| `directory`      | `question`, `clientFilesQuestion`, `clientFilesCourse`, `serverFilesCourse`, `courseExtensions` | `serverFilesCourse` | Parent directory to locate `file-name`.                                                                      |
+| `warn-undefined` | boolean                                                                                         | false               | Whether to print a warning when rendering templates with undefined variables. Useful for debugging.          |
+| `log-warnings`   | boolean                                                                                         | true                | Whether to log warnings if a rendered template contains elements which are not guaranteed to work correctly. |
+
+Inside the `pl-template` element, variables for use in rendering the template may be specified with a `pl-variable` tag. Each `pl-variable` tag can be used to define a variable from a file or with the contents of the tag (but not both). Note that recursion is **not** applied to data in `pl-variable`. The `pl-variable` tag supports the following attributes:
+
+| Attribute         | Type                                                                                            | Default             | Description                                                   |
+| ----------------- | ----------------------------------------------------------------------------------------------- | ------------------- | ------------------------------------------------------------- |
+| `name`            | string                                                                                          | -                   | Variable name to assign the data defined by this tag.         |
+| `file-name`       | string                                                                                          | -                   | File name to use if variable data is being taken from a file. |
+| `directory`       | `question`, `clientFilesQuestion`, `clientFilesCourse`, `serverFilesCourse`, `courseExtensions` | `serverFilesCourse` | Parent directory to locate `file-name`.                       |
+| `trim-whitespace` | boolean                                                                                         | true                | Whether to trim whitespace of data specified by this tag.     |
+
+#### Details
+
+Because of the way that elements are rendered in PrairieLearn, templates should only contain other decorative elements. In particular, **this element will not work correctly with elements that accept and/or grade student input.** When rendering a template, all entries from `data["params"]` are included as available variables and may be used when the template is rendered. Templates may also be used recursively.
```

Comment:
Is the problem that _this_ element won't work correctly, or is the problem that the graded element that is inside of it won't work correctly?

I guess it doesn't matter in the sense that in either case, you just don't want to put graded elements inside of this one. But it seems like we might as well say whatever is most correct if we can.

Reply:
The overall output just looks really weird, but you're correct that it's more precise to say that the graded element won't work as expected. Will change.

Reply:
Changed here: https://github.com/PrairieLearn/PrairieLearn/pull/6747/commits/6698f6249a1b8802227ccb4e8ea79618901306c1